admin/controllers/check-form.js

```diff
@@ -0,0 +1,269 @@
+const path = require('path')
+const fs = require('fs-extra')
+const CheckForm = require('../models/check-form')
+const checkFormService = require('../services/check-form.service')
+const checkFormDataService = require('../services/data-access/check-form.data.service')
+const checkWindowService = require('../services/check-window.service')
+
+/**
+ * Display landing page for 'test developer' role.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<void>}
+ */
+const getTestDeveloperHome = async (req, res, next) => {
+  res.locals.pageTitle = 'MTC for test development'
+  try {
+    req.breadcrumbs(res.locals.pageTitle)
+    res.render('test-developer/test-developer-home', {
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+/**
+ * Display initial 'upload and view' check forms page.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const uploadAndViewForms = async (req, res, next) => {
+  res.locals.pageTitle = 'Upload and view forms'
+  try {
+    const forms = await checkFormDataService.getActiveForms().sort({createdAt: -1}).exec()
+    let formData = forms.map(e => { return e.toJSON() })
+    const checkWindows = await checkWindowService.getCheckWindowsAssignedToForms()
+
+    formData.forEach(f => {
+      if (checkWindows[f._id]) {
+        f.checkWindows = checkWindows[f._id].map(cw => { return cw.toJSON() })
+      } else {
+        f.checkWindows = []
+      }
+    })
+
+    req.breadcrumbs(res.locals.pageTitle)
+    return res.render('test-developer/upload-and-view-forms', {
+      forms: formData,
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    return next(error)
+  }
+}
+
+/**
+ * Remove check form.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const removeCheckForm = async (req, res, next) => {
+  const id = req.params.formId
+  try {
+    const CheckWindow = await checkFormDataService.getActiveForm(id).exec()
+    if (!CheckWindow) {
+      return next(new Error(`Unable to find form.id [${id}]`))
+    }
+    // Un-assign check-form from any check-windows
+    const checkWindowsByForm = await checkWindowService.getCheckWindowsAssignedToForms()
+    if (checkWindowsByForm[CheckWindow._id]) {
+      // Array of CheckWindows models, each with a forms array
+      let modifiedCheckWindows = []
+      checkWindowsByForm[CheckWindow._id].forEach(cw => {
```

Comment:
I would expect these logic here to be added in a service as well.
It will make your tests way easier/simpler